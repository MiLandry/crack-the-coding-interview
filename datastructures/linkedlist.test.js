const LinkedList = require('./linkedlist');

let ll

beforeEach(() => {
  ll = new LinkedList()
  ll.add('a')
  ll.add('b')
  ll.add('c')

});

test('it adds', () => {
  expect(ll.head)
    .toMatchSnapshot()
})

test('it reads', () => {
  expect(ll.read(0))
  .toEqual('a')
})

test('it reads', () => {
  expect(ll.read(1))
  .toEqual('b')
})

test('it reads', () => {
  expect(ll.read(2))
  .toEqual('c')
})


test('it removes', () => {
  ll.remove(0)
  expect(ll.head)
    .toMatchSnapshot()
})


test('it removes 2', () => {
  ll.remove(1)
  expect(ll.head)
    .toMatchSnapshot()
})

test('it removes 3', () => {
  ll.remove(2)
  expect(ll.head)
    .toMatchSnapshot()
})
