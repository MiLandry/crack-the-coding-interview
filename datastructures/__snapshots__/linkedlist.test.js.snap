// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it adds 1`] = `
LinkedListNode {
  "data": "a",
  "next": LinkedListNode {
    "data": "b",
    "next": LinkedListNode {
      "data": "c",
      "next": null,
    },
  },
}
`;

exports[`it removes 1`] = `
LinkedListNode {
  "data": "b",
  "next": LinkedListNode {
    "data": "c",
    "next": null,
  },
}
`;

exports[`it removes 2 1`] = `
LinkedListNode {
  "data": "a",
  "next": LinkedListNode {
    "data": "c",
    "next": null,
  },
}
`;

exports[`it removes 3 1`] = `
LinkedListNode {
  "data": "a",
  "next": LinkedListNode {
    "data": "b",
    "next": null,
  },
}
`;
